grove/ising/ising_qaoa.py

```diff
@@ -0,0 +1,142 @@
+
+"""
+Finding the minimum energy for an Ising problem by QAOA.
+"""
+import pyquil.api as api
+from grove.pyqaoa.qaoa import QAOA
+from pyquil.paulis import PauliSum, PauliTerm
+from scipy.optimize import minimize
+import numpy as np
+
+CXN = api.SyncConnection()
+
+
+def energy_value(h, J, sol):
+    """
+    Obtain energy of an Ising solution for a given Ising problem (h,J).
+
+    :param h: External magnectic term of the Ising problem. List.
+    :param J: Interaction term of the Ising problem. Dictionary.
+    :param sol: Ising solution. List.
+
+    """
+    ener_ising = 0
+    for elm in J.keys():
+        if elm[0] == elm[1]:
+            ener_ising += J[elm] * int(sol[elm[0]])
+        else:
+            ener_ising += J[elm] * int(sol[elm[0]]) * int(sol[elm[1]])
+    for i in range(len(h)):
+        ener_ising += h[i] * int(sol[i])
+    return ener_ising
+
+
+def print_fun(x):
+    print(x)
+
+
+def ising_trans(x):
```

Comment:
Instead of having this function floating around you could just absorb it in the calculation of the energies?

Reply:
This function is also used in the ising function, this is the reason I left it there.